Vistas creadas junto con el controlador.

diegoOG09 opened the pull request:
Vistas creadas junto con el controlador.

- Vista creada de la sección principal de "Chambas".
-  Vista creada de cuando se accede a una determinada "chamba".
- El controlador fue modificado para mostrar algunos filtros de ejemplo en la vista.
- Se han añadido rutas al archivo 'routes.rb' para poder ver los cambios en el
  servidor local.

Co-authored-by: diegoOG09 <xdiegoemmax@outlook.com>
Co-authored-by: armanirmr <armanirmr@gmail.com>
Co-authored-by: YxMARCKxY <mramirez73@ucol.mx>
Co-authored-by: rene3255 <ranchoeltecal@gmail.com>
Co-authored-by: BrandonVG <bvadillo0@ucol.mx>

![image](https://user-images.githubusercontent.com/63070877/151888911-9ff4086c-070b-4bd4-be90-e67fa9d65494.png)

<!--
     For Work In Progress Pull Requests, please use the Draft PR feature,
     see https://github.blog/2019-02-14-introducing-draft-pull-requests/ for further details.

     For a timely review/response, please avoid force-pushing additional
     commits if your PR already received reviews or comments.

     Before submitting a Pull Request, please ensure you've done the following:
     - 📖 Read the Forem Contributing Guide: https://github.com/forem/forem/blob/main/CONTRIBUTING.md#create-a-pull-request.
     - 📖 Read the Forem Code of Conduct: https://github.com/forem/forem/blob/main/CODE_OF_CONDUCT.md.
     - 👷‍♀️ Create small PRs. In most cases this will be possible.
     - ✅ Provide tests for your changes.
     - 📝 Use descriptive commit messages.
     - 📗 Update any related documentation and include any relevant screenshots.

     NOTE: Pull Requests from forked repositories will need to be reviewed by
     a Forem Team member before any CI builds will run. Once your PR is approved
     with a `/ci` reply to the PR, it will be allowed to run subsequent builds without
     manual approval.
-->

## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

## Related Tickets & Documents

## QA Instructions, Screenshots, Recordings

_Please replace this line with instructions on how to test your changes, a note
on the devices and browsers this has been tested on, as well as any relevant
images for UI changes._

### UI accessibility concerns?

_If your PR includes UI changes, please replace this line with details on how
accessibility is impacted and tested. For more info, check out the
[Forem Accessibility Docs](https://developers.forem.com/frontend/accessibility)._

## Added/updated tests?

- [ ] Yes
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [ ] I need help with writing tests

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

- [ ] I've updated the [Developer Docs](https://developers.forem.com) or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
- [ ] This PR changes the Forem platform and our documentation needs to be
      updated. I have filled out the
      [Changes Requested](https://github.com/forem/admin-docs/issues/new?assignees=&labels=&template=changes_requested.md)
      issue template so Community Success can help update the Admin Docs
      appropriately.
- [ ] I've updated the README or added inline documentation
- [ ] I've added an entry to
      [`CHANGELOG.md`](https://github.com/forem/forem/tree/main/CHANGELOG.md)
- [ ] I will share this change in a [Changelog](https://forem.dev/t/changelog)
      or in a [forem.dev](http://forem.dev) post
- [ ] I will share this change internally with the appropriate teams
- [ ] I'm not sure how best to communicate this change and need help
- [ ] This change does not need to be communicated, and this is why not: _please
      replace this line with details on why this change doesn't need to be
      shared_

## [optional] Are there any post deployment tasks we need to perform?

## [optional] What gif best describes this PR or how it makes you feel?

![alt_text](gif_link)

Commit:
Views created, "Chambas" controller created and edited and draft database.

- Created view of the main section of "Chambas".
- Created view of the "Glossary" section, although some details are still to be added.
- Created view of when a particular "chamba" is accessed.
- The controller was modified to show some example filters in the view.
- Added routes to the 'routes.rb' file to be able to view changes on the
  local server.

Co-authored-by: diegoOG09 <xdiegoemmax@outlook.com>
Co-authored-by: armanirmr <armanirmr@gmail.com>
Co-authored-by: YxMARCKxY <mramirez73@ucol.mx>
Co-authored-by: rene3255 <ranchoeltecal@gmail.com>
Co-authored-by: BrandonVG <bvadillo0@ucol.mx>

EduardoGHdez commented on Gemfile.lock:
```diff
@@ -63,6 +63,7 @@ GEM
       activerecord (= 6.1.4.4)
       activesupport (= 6.1.4.4)
       marcel (~> 1.0.0)
+      mini_magick
```
Me pregunto si agregaste mini_magick en el `Gemfile` para que el proyecto pudiera correr en tu ambiente local, o si esto se agrego solo 🤔 

Si no es necesario, podriamos remover esto del commit? 😄

EduardoGHdez commented on app/assets/stylesheets/chambas.scss:
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the chambas controller here.
+// They will automatically be included in application.css.
+// You can use Sass (SCSS) here: https://sass-lang.com/
```
Si no necesitamos este fichero, podemos removerlo? 😄

EduardoGHdez commented on app/assets/stylesheets/components/cards.scss:
```diff
@@ -42,3 +42,104 @@
+  }
+}
+
+i {
+  margin-right: 0.3rem;
+}
```
Hay que tener cuidado al usar elementos en CSS, Podriamos ser mas especificos con este selector?
Esto podria alterar el estilo de todos los elementos `<i>` en el proyecto

EduardoGHdez commented on app/controllers/chambas_controller.rb:
```diff
@@ -0,0 +1,6 @@
+class ChambasController < ApplicationController
+  def chambas
+    @filters = %w[Mexico Web-Developer Programación Issues]
```
Podemos mover esto a una constante, en incluso al modelo de Chambas? 😄

EduardoGHdez commented on app/views/pages/glosario.html.erb:
```diff
@@ -0,0 +1,28 @@
+<% title "Glosario" %>
+
+<div class="crayons-layout crayons-layout--2-cols crayons-layout--2-cols--drop-right-left">
+  <div class="sidebar-wrapper sidebar-wrapper-left crayons-layout__sidebar-left" id="sidebar-wrapper-left">
+    <aside class="side-bar" aria-label="Primary sidebar">
+      <%= render "layouts/main_nav" %>
+    </aside>
+  </div>
+  <main class="crayons-layout__content">
+    <h1 class="">Glosario</h1>
+    <div class="listings-columns" id="listings-results">
+      <div class="single-listing crayons-card text-styles text-padding">
+        <p>
+          Política publica: Acciones del gobierno (o documentos que generen o propongan acciones del gobierno), que surgen de
+          un proceso de diagnóstico y análisis de factibilidad para la atención efectiva de problemas públicos,
+          en donde la definición de problemas y soluciones se lleva a cabo con la participación ciudadana.
+        </p>
+      </div>
+      <div class="single-listing crayons-card text-styles text-padding">
+        <p>
+          Política publica: Acciones del gobierno (o documentos que generen o propongan acciones del gobierno), que surgen de
+          un proceso de diagnóstico y análisis de factibilidad para la atención efectiva de problemas públicos,
+          en donde la definición de problemas y soluciones se lleva a cabo con la participación ciudadana.
+        </p>
+      </div>
+    </div>
+  </main>
+</div>
```
Estoy confundido, esta seccion estaba en el diseño?

YxMARCKxY replied:
Esta en los [mockups ](https://www.figma.com/file/t2L7DdgcAls1CQt4Ie4pNv/Forem-Views?node-id=0%3A1) es el glosario, o no es necesario?

EduardoGHdez replied:
ya veo!

Ignoren eso por ahora, y limitemos este PR a la seccion de Chambas unicamente 👍🏽

EduardoGHdez commented on config/database.yml:
```diff
@@ -27,11 +27,13 @@ default: &default
   database: <%= ENV.fetch("DATABASE_NAME", "Forem_development") %>
   variables:
       statement_timeout: <%= ENV.fetch("STATEMENT_TIMEOUT", 10_000).to_i %>
+  username: forem-test
+  password: forem-test
 
+  #url: <%= ENV.fetch("DATABASE_URL", "postgres:///Forem_development") %>
```
No es necesario agregar estos cambios al proyecto, ya que la configuracion puede cambiar segun preferencias personales.

EduardoGHdez commented on db/schema.rb:
```diff
@@ -14,7 +14,6 @@
 
   # These are extensions that must be enabled in order to support this database
   enable_extension "citext"
-  enable_extension "pg_stat_statements"
```
Este cambio no es necesario.

Chistosamente lei de esta extension hoy 😂 

https://www.citusdata.com/blog/2019/02/08/the-most-useful-postgres-extension-pg-stat-statements/

por si gustas ver para que funciona, sin embargo, no es necesario que la habilites en tu maquina, solo evitemos commitear cambios al schema cuando no son necesarios 😄

EduardoGHdez commented on db/schema.rb:
```diff
@@ -1224,7 +1223,7 @@
     t.datetime "last_comment_at", default: "2017-01-01 05:00:00"
     t.datetime "last_followed_at"
-    t.datetime "last_moderation_notification", default: "2017-01-01 05:00:00"
+    t.datetime "last_moderation_notification", default: "2017-01-01 06:00:00"
```
Este es raro, no se por que se habra cambiado 🤔 , pero igual podemos quitarlo del PR, pls? 😄 🙏🏽

EduardoGHdez commented on spec/helpers/chambas_helper_spec.rb:
```diff
@@ -0,0 +1,15 @@
+require "rails_helper"
+
+# Specs in this file have access to a helper object that includes
+# the ChambasHelper. For example:
+#
+# describe ChambasHelper do
+#   describe "string concat" do
+#     it "concats two strings with spaces" do
+#       expect(helper.concat_strings("this","that")).to eq("this that")
+#     end
+#   end
+# end
+RSpec.describe ChambasHelper, type: :helper do
+  pending "add some examples to (or delete) #{__FILE__}"
+end
```
Removamos este modulo y su respectivo test, si es que hace sentido 😄 

O hay algun plan para agregar estos helpers?

EduardoGHdez commented on app/controllers/chambas_controller.rb:
```diff
@@ -0,0 +1,6 @@
+class ChambasController < ApplicationController
+  def chambas
```
```suggestion
  def index
```

Intentemos seguir las conveciones de REST

EduardoGHdez commented on spec/requests/chambas_spec.rb:
```diff
@@ -0,0 +1,7 @@
+
+RSpec.describe "Chambas", type: :request do
+  describe "GET /index" do
+    pending "add some examples (or delete) #{__FILE__}"
```
Podemos agregar algun test? 😄

EduardoGHdez commented on vendor/cache/acts_as_follower-06393d3693a1/acts_as_follower.gemspec:
```diff
@@ -8,16 +8,16 @@ Gem::Specification.new do |s|
   s.require_paths = ["lib".freeze]
   s.authors = ["Tom Cocca".freeze]
-  s.date = "2021-04-09"
+  s.date = "2022-01-25"
```
Interesante 🤔 

Como se dio este cambio?
Creo que fue por que forzamos la instalacion de las gemas... podemos intentar deshacer este cambio y ver si sigue funcionando?

EduardoGHdez reviewed (commented):
Buen avance! 🔥 

Deje unos cuantos comentarios, pero esto se ve genial 🙌🏽 
Gracias por la contribución 💯

Commit:
We resolved the observations from the last PR

Co-authored-by: diegoOG09 <xdiegoemmax@outlook.com>
Co-authored-by: armanirmr <armanirmr@gmail.com>
Co-authored-by: YxMARCKxY <mramirez73@ucol.mx>
Co-authored-by: rene3255 <ranchoeltecal@gmail.com>
Co-authored-by: BrandonVG <bvadillo0@ucol.mx>

diegoOG09 committed:
Chambas usuario y admin OK|Create chamba Admin OK

diegoOG09 committed:
Migrations added to Chambas db

- We add certain fields in the model of chambas as: stall, approve

Co-authored-by: diegoOG09 <xdiegoemmax@outlook.com>
Co-authored-by: armanirmr <armanirmr@gmail.com>
Co-authored-by: YxMARCKxY <mramirez73@ucol.mx>
Co-authored-by: rene3255 <ranchoeltecal@gmail.com>
Co-authored-by: BrandonVG bvadillo0@ucol.mx
Co-authored-by: IrenehMDev <ireneh.dev_03@icloud.com>

diegoOG09 committed:
Migrations modified

- we put together some migration to chambas in only one  by esthetic and good practice

Co-authored-by: armanirmr <armanirmr@gmail.com>
Co-authored-by: YxMARCKxY <mramirez73@ucol.mx>
Co-authored-by: rene3255 <ranchoeltecal@gmail.com>
Co-authored-by: BrandonVG bvadillo0@ucol.mx

diegoOG09 committed:
CRUD views and admin Chambas created

-We added the new section of chambas within the content manager.
-We added the list within the content manager in chambas.
-We create the view to see a single job.
-We create the crud within the administration section in content manager.
-We add some js scripts to manipulate the information of the chambas within the form when making updates of specific chambas.
-We add some styles to the cards of the chambas.
-We added some features in the content manager section for jobs such as:
-Ban chamba posts
-Update the expiration date of the jobs
-Approve the publications of new chambas

Co-authored-by: armanirmr armanirmr@gmail.com
Co-authored-by: YxMARCKxY mramirez73@ucol.mx
Co-authored-by: rene3255 ranchoeltecal@gmail.com
Co-authored-by: BrandonVG bvadillo0@ucol.mx
